### PR TITLE
Upgrading Azure storage dependency to 7.0.0

### PR DESCRIPTION
**Pull request by xavierdecoster**
In preparation of some cloud service project changes.
Azure Tools 2.9 by default upgrade web/worker role project dependencies to at least use storage sdk v6.2.0.

This change upgrades to v7.0 at once in a separate PR.

cc @maartenba @skofman1

### CODE REVIEW

File: tests/CatalogTests/InstallDataBrowser.cs [hunk 1]
```diff
@@ -28,9 +25,9 @@ public static void Test0()
             blob.Properties.ContentType = "text/html";
             blob.Properties.CacheControl = "no-store";
 
-            blob.UploadFromFile("DataBrowser\\index.html", System.IO.FileMode.Open);
+            blob.UploadFromFile("DataBrowser\\index.html");
```
Comment (xavierdecoster) [inline tests/CatalogTests/InstallDataBrowser.cs]: This simply addresses a breaking change introduced in Azure storage SDK v7.

File: src/NuGet.Services.BasicSearch/Web.config [hunk 2]
```diff
@@ -4,6 +4,15 @@
   http://go.microsoft.com/fwlink/?LinkId=169433
   -->
 <configuration>
+  <system.diagnostics>
+    <trace>
+      <listeners>
+        <add type="Microsoft.WindowsAzure.Diagnostics.DiagnosticMonitorTraceListener, Microsoft.WindowsAzure.Diagnostics, Version=2.8.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" name="AzureDiagnostics">
```
Comment (maartenba) [inline src/NuGet.Services.BasicSearch/Web.config]: Any reason this change is part of the PR?

**Comment by maartenba** [comment]
Have you tried the `AzureDirectory` to see if it likes the binding redirect?

Otherwise LGTM :shipit:

File: src/NuGet.Services.BasicSearch/Web.config (hunk 2 above)
Comment (xavierdecoster) [inline src/NuGet.Services.BasicSearch/Web.config]: This one was added by the NuGet package upgrade. Seems Azure diagnostics changed configuration settings again? Happy to remove it if not needed.

Comment (maartenba) [inline src/NuGet.Services.BasicSearch/Web.config]: Yank it :-)

File: src/Catalog/NuGet.Services.Metadata.Catalog.csproj [hunk 3]
```diff
@@ -60,6 +60,10 @@
       <HintPath>..\..\packages\json-ld.net.1.0.4\lib\net40-Client\JsonLD.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.Azure.KeyVault.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (skofman1) [inline src/Catalog/NuGet.Services.Metadata.Catalog.csproj]: do we need this dependency?

File: src/Ng/Ng.csproj [hunk 4]
```diff
@@ -117,6 +117,10 @@
       <HintPath>..\..\packages\Microsoft.ApplicationInsights.2.1.0-beta3\lib\net45\Microsoft.ApplicationInsights.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.Azure.KeyVault.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (skofman1) [inline src/Ng/Ng.csproj]: same

File: src/Ng/packages.config [hunk 5]
```diff
@@ -6,6 +6,7 @@
   <package id="Lucene.Net.Contrib" version="3.0.3" targetFramework="net452" />
   <package id="Lucene.Net.Store.Azure" version="3.0.5553.21100" targetFramework="net452" />
   <package id="Microsoft.ApplicationInsights" version="2.1.0-beta3" targetFramework="net452" />
+  <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net452" />
```
Comment (skofman1) [inline src/Ng/packages.config]: same

File: src/NuGet.Indexing/packages.config [hunk 6]
```diff
@@ -4,6 +4,7 @@
   <package id="Lucene.Net" version="3.0.3" targetFramework="net452" />
   <package id="Lucene.Net.Contrib" version="3.0.3" targetFramework="net452" />
   <package id="Lucene.Net.Store.Azure" version="3.0.5553.21100" targetFramework="net452" />
+  <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net452" />
```
Comment (skofman1) [inline src/NuGet.Indexing/packages.config]: same

File: src/NuGet.Services.BasicSearch/NuGet.Services.BasicSearch.csproj [hunk 7]
```diff
@@ -88,6 +88,10 @@
       <HintPath>..\..\packages\Microsoft.ApplicationInsights.RuntimeTelemetry.0.12.0-build17386\lib\net45\Microsoft.ApplicationInsights.Extensibility.RuntimeTelemetry.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.Azure.KeyVault.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (skofman1) [inline src/NuGet.Services.BasicSearch/NuGet.Services.BasicSearch.csproj]: same

**Comment by skofman1** [comment]
Some unnecessary references to KeyVault in all projects

File: src/Catalog/NuGet.Services.Metadata.Catalog.csproj (hunk 3 above)
Comment (xavierdecoster) [inline src/Catalog/NuGet.Services.Metadata.Catalog.csproj]: Yep, see https://www.nuget.org/packages/WindowsAzure.Storage/7.0.0 (listed as a dependency :))

**Comment by xavierdecoster** [comment]
@skofman1 it's a dependency of WindowsAzure.Storage v7.0.0

**Comment by skofman1** [comment]
Unexpected :) :ship:

**Comment by xavierdecoster** [comment]
synced offline with @maartenba, verified that binding redirects are non-issue

**Pull request merged**